core/server/common/src/main/java/alluxio/master/journal/AsyncJournalWriter.java

```diff
@@ -99,49 +146,161 @@ public long appendEntry(JournalEntry entry) {
   }
 
   /**
-   * Flushes and waits until the specified counter is flushed to the journal. If the specified
-   * counter is already flushed, this is essentially a no-op.
-   *
-   * @param targetCounter the counter to flush
+   * Closes the async writer.
+   * PS: It initiates but does not wait for outstanding entries to be flushed.
```

Comment:
Made some improvements for attempting to write entries before bailing. 
However I believe strong guarantees should be enforced by calling ::flush() because there may be concurrent writers while closing it.